### PR TITLE
Add sample for Selective Consumer

### LINKED ISSUE #90: Create Sample for Selective Consumer pattern
Create Sample for [Selective Consumer](https://www.enterpriseintegrationpatterns.com/patterns/messaging/MessageSelector.html) pattern

### PR DESCRIPTION
**Pull request by ShammiL**
Fixes #90

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: selective-consumer/selective-consumer.bal [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+enum RequestType {
+    REQUIRED,
+    URGENT
+};
+
+final CsvRecord[] csvContent = [];
+final string csvFilePath = "./resources/orderRequests.csv";
+
+final http:Client shopify = check new ("http://BlackwellsBooks.myshopify.com.balmock.io");
```
Comment (manuranga) [inline selective-consumer/selective-consumer.bal]: lower case.

File: selective-consumer/selective-consumer.bal [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+public function main(string category) returns error? {
+            InventoryResponse inventories = check shopify->/admin/api/graphql\.json.post({
```
Comment (manuranga) [inline selective-consumer/selective-consumer.bal]: indention.

File: selective-consumer/selective-consumer.bal [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                query: string `query { products(product_type: "${category}") { name productsCount } }`
+            });
+            _ = from Inventory inventory in inventories.products
+                select createCsv(inventory);
```
Comment (manuranga) [inline selective-consumer/selective-consumer.bal]: better to foreach

File: selective-consumer/selective-consumer.bal [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+}
+
+function createCsv(Inventory product) returns error? {
+    if product.productsCount < 25 && product.productsCount > 10 {
+        csvContent.push({name: product.name, requestType: REQUIRED});
+    } else if product.productsCount < 10 {
+        csvContent.push({name: product.name, requestType: URGENT});
+    }
```
Comment (LakshanWeerasinghe) [inline selective-consumer/selective-consumer.bal]:
```suggestion
    if product.productsCount < 10 {
        csvContent.push({name: product.name, requestType: URGENT});
    } else if product.productsCount < 25 {
        csvContent.push({name: product.name, requestType: REQUIRED});
    }
```

File: selective-consumer/selective-consumer.bal [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    RequestType requestType;
+|};
+
+enum RequestType {
+    REQUIRED,
+    URGENT
+};
+
+final graphql:Client shopify = check new ("http://blackwellsbooks.myshopify.com.balmock.io");
+// final graphql:Client shopify = check new ("localhost:8080");
```
Comment (LakshanWeerasinghe) [inline selective-consumer/selective-consumer.bal]: remove this comment

File: selective-consumer/selective-consumer.bal [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+public function main(string category) returns error? {
+    final string csvFilePath = "./resources/orderRequests.csv";
+    CsvRecord[] csvContent = [];
```
Comment (LakshanWeerasinghe) [inline selective-consumer/selective-consumer.bal]: move this to line 31.

File: selective-consumer/selective-consumer.bal [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import ballerina/graphql;
+import ballerina/io;
+
+type InventoryResponse record {|
+    record {|Inventory[] products;|} data;
+|};
+
+type Inventory record {
```
Comment (LakshanWeerasinghe) [inline selective-consumer/selective-consumer.bal]: make this a closed record.

File: selective-consumer/selective-consumer.bal [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+};
+
+final graphql:Client shopify = check new ("http://blackwellsbooks.myshopify.com.balmock.io");
+
+public function main(string category) returns error? {
+    string csvFilePath = "./resources/orderRequests.csv";
+    string document = string `{ products(productType: "${category}") { name, productsCount } } `;
+    InventoryResponse inventories = check shopify->execute(document);
+    CsvRecord[] csvContent = [];
+    foreach Inventory {name, productsCount} in inventories.data.products {
```
Comment (manuranga) [inline selective-consumer/selective-consumer.bal]:
```suggestion
    foreach var {name, productsCount} in inventories.data.products {
```

**Review by manuranga — APPROVED** [review]

**Pull request merged**